### PR TITLE
Update dockerfile for adservice to use different base images

### PR DESCRIPTION
**Pull request by svrnm**
# Changes

Minor change for the adservice: Similar to frauddetectionservice this makes use of the gradle image to save some time downloading gradle (since both use the same gradle image then this even saves some extra time downloading the image). It also replaces the jdk with a jre image for the final image, saves some ~190MB (518 down to 328)

### CODE REVIEW

**Comment by cartersocha** [comment]
10s build time improvement and same size reduction. lgtm

**Review by cartersocha — APPROVED** [review]
🚀

File: src/adservice/Dockerfile [hunk 1]
```diff
@@ -12,26 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM eclipse-temurin:17-jdk AS builder
+FROM gradle:7-jdk17 AS builder
```
Comment (jack-berg) [inline src/adservice/Dockerfile:15]: I don't recommend using a the gradle wrapper for local builds / development and using a different version of gradle for the docker build.

Instead, you can speed up the docker build by ordering the commands such that things that change infrequently like the gradle wrapper and the dependencies are cached in intermediate layers:
```
FROM eclipse-temurin:17-jdk AS builder

WORKDIR /usr/src/app/

COPY ./src/adservice/gradlew* ./src/adservice/settings.gradle* ./src/adservice/build.gradle .
COPY ./src/adservice/gradle ./gradle

RUN ./gradlew
RUN ./gradlew downloadRepos

COPY ./src/adservice/ ./
COPY ./pb/ ./proto
RUN ./gradlew installDist -PprotoSourceDir=./proto

# -----------------------------------------------------------------------------

FROM eclipse-temurin:17-jre
```

Comment (svrnm) [reply src/adservice/Dockerfile:15]: Makes sense, let me fix this here (and with the frauddetectionservice, where I did the same)

**Review by austinlparker — APPROVED** [review]

File: src/adservice/Dockerfile (hunk 1 above)
Comment (svrnm) [reply src/adservice/Dockerfile]: Updated it accordingly

**Comment by svrnm** [comment]
followed @jack-berg's comment and changed the builder once again

**Comment by cartersocha** [comment]
Thanks @svrnm and @jack-berg 🔥

**Pull request merged**